llarp/router/router.cpp

```diff
@@ -859,7 +861,17 @@ namespace llarp
       connectToNum = strictConnect;
     }
 
-    if (connected < connectToNum)
+    if (looksDeregistered)
+    {
+      // kill all sessions that are open because we think we are deregistered
+      _linkManager.ForEachPeer([](auto* peer) {
+        if (peer)
+          peer->Close();
+      });
```

Comment:
Isn't this a bit harsh?  If I have a path going through router X, and X gets decommissioned, this is going to immediately kill all the paths through it, which is going to result in client-side failures.  Could we do something a bit more graceful here, such as putting this behind a timer so that we kill all paths only once we have looked deregistered for more than xx minutes (where xx = whatever maximum path validity is, I think 20min?).